contracts/lazymint/ERC721LazyMint.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+import "@manifoldxyz/creator-core-solidity/contracts/core/IERC721CreatorCore.sol";
+import "@manifoldxyz/libraries-solidity/contracts/access/AdminControl.sol";
+import "@manifoldxyz/creator-core-solidity/contracts/extensions/ICreatorExtensionTokenURI.sol";
+
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/interfaces/IERC165.sol";
+// import "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+import "./IERC721LazyMint.sol";
+
+/**
+ * Lazy mint with whitelist ERC721 tokens
+ */
+contract ERC721LazyMint is IERC165, IERC721LazyMint, ICreatorExtensionTokenURI, ReentrancyGuard {
+  mapping(address => uint) public listingCounts;
+  mapping(address => mapping(uint => Listing)) public listings;
+  mapping(address => mapping(uint => uint)) public mintsPerListing;
+  mapping(address => mapping(uint => mapping(address => uint))) public mintsPerWallet;
+  mapping(address => mapping(uint => uint)) public tokenListings;
+
+  function supportsInterface(bytes4 interfaceId) public view virtual override(IERC165) returns (bool) {
+    return interfaceId == type(IERC721LazyMint).interfaceId ||
+      interfaceId == type(ICreatorExtensionTokenURI).interfaceId ||
+      interfaceId == type(IERC165).interfaceId;
+  }
+
+  modifier creatorAdminRequired(address creatorContractAddress) {
+    // This is shared, not single-creator. So we must ensure that the initializer is an admin of the contract
+    AdminControl creatorCoreContract = AdminControl(creatorContractAddress);
+    require(creatorCoreContract.isAdmin(msg.sender), "Wallet is not an administrator for contract");
+    _;
+  }
+
+  // Initialize
+  function initializeListing(
```

Comment:
Wondering if we should have an update function that sets all the parameters again.

In the app, if you edit the draft and publish again, we would then need to have logic to see which parameters changed and use the right functions to update them potentially having more than one transaction (e.g if you update the token and start date you'll need to run 2 txs vs just the initialize function called the first time)